core/lib/server/pub.ts

```diff
@@ -1620,13 +1669,33 @@ export async function getPubsWithRelatedValues<Options extends GetPubsWithRelate
 								)
 						)
 					)
-					.$if(Boolean(props.pubId), (qb) => qb.where("pubs.id", "=", props.pubId!))
-					.$if(Boolean(props.stageId), (qb) =>
-						qb.where("PubsInStages.stageId", "=", props.stageId!)
+					.$if(!!props.pubId, (qb) => qb.where("pubs.id", "=", props.pubId!))
+					.$if(!!props.pubIds && props.pubIds.length > 0, (qb) =>
+						qb.where("pubs.id", "in", props.pubIds!)
 					)
-					.$if(Boolean(props.pubTypeId), (qb) =>
+					// stage filter
+					// we need to do these checks separately bc just bc you are only allowed to see
+					// some stages, doesn't mean you _want_ to see all stages
+					// props.stageId is the list of stages the user wants to see
+					// allowedStages is the list of stages the user is allowed to see
+					.$if(Boolean(props.stageId?.length), (qb) =>
+						qb.where((eb) => stagesWhere(eb, props.stageId!, "PubsInStages.stageId"))
+					)
+					.$if(Boolean(allowedStages?.length), (qb) =>
+						qb.where((eb) => stagesWhere(eb, allowedStages!, "PubsInStages.stageId"))
+					)
```

Comment:
note: these two checks could be combined, but i find the logic would become too complex to follow. 
example: if `props.stageId` = [ 'Submission', 'Evaluation' ], and `allowedStages = ['Evaluation']`, the `where` could just be `in [ 'Evaluation' ]`. We could always filter `props.stageId` by `allowedStages`, and then use the smallest set.
But I find all this logic to become very complex, and im fine with maybe sacrificing a bit of performance by maybe doing two `where in` checks.

Same goes for the pubTypes below